components/Navbar.jsx

```diff
@@ -6,39 +6,30 @@ import { FaGithub, FaLinkedinIn } from "react-icons/fa";
 import { BsFillPersonLinesFill } from "react-icons/bs";
 import { useRouter } from "next/router";
 
-const Navbar = () => {
+const Navbar = ({ isDark }) => {
   const [nav, setNav] = useState(false);
   const [shadow, setShadow] = useState(false);
   const [navBg, setNavBg] = useState("#ecf0f3");
   const [linkColor, setLinkColor] = useState("#1f2937");
   const router = useRouter();
 
   useEffect(() => {
-    if (
-      router.asPath === "/White" ||
-      router.asPath === "/White2" ||
-      router.asPath === "/White3" ||
-      router.asPath === "/White4"
-    ) {
+    if (isDark) {
       setNavBg("transparent");
       setLinkColor("#ecf0f3");
     } else {
       setNavBg("#ecf0f3");
       setLinkColor("#1f2937");
     }
-  }, [router]);
+  }, [isDark]);
 
   const handleNav = () => {
     setNav(!nav);
   };
 
   useEffect(() => {
     const handleShadow = () => {
-      if (window.scrollY >= 90) {
-        setShadow(true);
-      } else {
-        setShadow(false);
-      }
+      setShadow(window.scrollY >= 90);
     };
     window.addEventListener("scroll", handleShadow);
   }, []);
```

Comment:
```
  const checkScrollTop = () => {  
    setShadow(window.scrollY >= 90);
  };
  
  useEffect(() => {
    window.addEventListener('scroll', checkScrollTop);
    return () => window.removeEventListener('scroll', checkScrollTop);
  }, []);
  ```

Reply:
Although it's a navbar component, and maybe it will never be removed from the UI; it's always a good practice to handle removing window event listeners after the component delete.